pilot/pkg/model/xds_cache.go

```diff
@@ -207,11 +217,12 @@ func (l *lruCache) Add(entry XdsCacheEntry, token CacheToken, value *any.Any) {
 		// first write, or we forgot to call Get before.
 		return
 	}
-	if features.EnableUnsafeAssertions {
+	if l.enableAssertions {
 		if toWrite.token == 0 {
+			xdsCacheInvalidUpdate.Increment()
```

Comment:
if enable assertion is intended to be used only by test, then it is not meaning record this error.

Reply:
This is used to measure invalid calls when it is enabled - but since any way we are panicking below may be not needed?

@howardjohn what do you think? Would it be useful? Should I just close this PR?